tools/scripts/rebuildAnimationLibraryManifest.rb

```diff
@@ -0,0 +1,217 @@
+        alias_map[aliaz] = ((alias_map[aliaz] || []) + [name]).uniq.sort
+      end
+    end
+
+    alias_map.each {|k, v| verbose "#{bold k}: #{v.join(', ')}"} if @options[:verbose]
+
+    info "Mapped #{alias_map.size} aliases."
+
+    # Write result to file
+    File.open(DEFAULT_OUTPUT_FILE, 'w') do |file|
+      file.write(JSON.pretty_generate({
+          'metadata': animation_metadata_by_name,
+          'aliases': alias_map
+      }))
+    end
+    info <<-EOS.unindent
+      Manifest written to #{DEFAULT_OUTPUT_FILE}.
+
+        #{dim 'd[ o_0 ]b'}
+    EOS
+
+  # Report any issues while talking to S3 and suggest most likely steps for fixing it.
+  rescue Aws::Errors::ServiceError => service_error
+    warn service_error.inspect
+    warn <<-EOS.unindent
+
+      #{bold 'There was an error talking to S3.'}  Make sure you have credentials set using one of:
+
+        * aws_access_key and aws_secret_key in your locals.yml
+        * ENV['AWS_ACCESS_KEY_ID'] and ENV['AWS_SECRET_ACCESS_KEY']
+        * ~/.aws/credentials
+
+      #{dim 'See http://docs.aws.amazon.com/sdkforruby/api/Aws/S3/Client.html for more details.'}
+    EOS
+  end
+
+  def verbose(s)
+    puts(s) if @options[:verbose]
+  end
+
+  def info(s)
+    puts(s) unless @options[:quiet]
+  end
+
+  def warn(s)
+    puts(s)
+  end
+end
+
+# Given the complete body of a PNG file, returns hash of source dimensions:
+# {"x": _, "y": _}
+def dimensions_from_png(png_body)
+  # Read the first eight bytes of the IHDR Chunk, which must always be the
+  # first chunk of the PNG file.
+  #
+  # PNG Header takes 8 bytes (0x00-0x07)
+  # IHDR chunk length takes 4 bytes (0x08-0x0b)
+  # IHDR chunk type code takes 4 bytes (0x0c-0x0f)
+  # IHDR chunk data begins at 0x10
+  #
+  # The IHDR chunk begins with
+  # 4 bytes for width (0x10-0x13) followed by
+  # 4 bytes for height (0x14-0x17)
+  #
+  # PNG File Structure
+  # http://www.libpng.org/pub/png/spec/1.2/PNG-Structure.html
+  # IHDR Chunk Layout
+  # http://www.libpng.org/pub/png/spec/1.2/PNG-Chunks.html#C.IHDR
+  dimensions = png_body[0x10..0x18].unpack('NN')
+  {'x': dimensions[0], 'y': dimensions[1]}
+end
+
+# Utility to strip consistent leading whitespace from heredoc strings, allowing
+# you to format your code more readably.
+#
+# Usage:
+#   <<-DOC.unindent
+#     Some text or other that I want to start in column 1.
+#         An actually indented line.
+#     This line still unindented.
+#   DOC
+#
+# from http://stackoverflow.com/a/5638187/5000129
+class String
+  # Strip leading whitespace from each line that is the same as the
+  # amount of whitespace on the least-indented line of the string.
+  def unindent
+    gsub /^#{scan(/^[ \t]+/).min_by(&:length)}/, ''
+  end
+end
+
+# Terminal style utilities for fashionable output
+def stylize(text, color_code)
+  "\e[#{color_code}m#{text}\e[0m"
+end
+
+def bold(text); stylize(text, 1); end
+def dim(text); stylize(text, 2); end
+def underline(text); stylize(text, 4); end
```

Comment:
I copied this stuff from the dotd script.  I guess I should extract a cli_helpers.rb file somewhere, huh?